domain/src/main/java/org/oppia/domain/analytics/EventStorageHelper.kt

```diff
@@ -0,0 +1,97 @@
+package org.oppia.domain.analytics
+
+import androidx.lifecycle.LiveData
+import org.oppia.app.model.EventLog
+import org.oppia.app.model.EventLogs
+import org.oppia.data.persistence.PersistentCacheStore
+import org.oppia.util.data.AsyncResult
+import org.oppia.util.data.DataProviders
+import org.oppia.util.logging.Logger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class EventStorageHelper @Inject constructor(
+  private val cacheStoreFactory: PersistentCacheStore.Factory,
+  private val dataProviders: DataProviders,
+  private val logger: Logger
+) {
+  private val eventLogStore =
+    cacheStoreFactory.create("event_logs", EventLogs.getDefaultInstance())
+
+  fun addEventLog(eventLog: EventLog) {
+    eventLogStore.storeDataAsync(updateInMemoryCache = true) {
```

Comment:
Can we check for the storage limit here itself before storing a new record? i.e. check storage, if its greater than the limit then delete a log. Then add this log

Reply:
Yes, working on it.